### PR TITLE
developer_manual/design: Update mail* pages

### PR DESCRIPTION
**Pull request by sirainen**

### CODE REVIEW

File: source/developer_manual/design/mailbox_list.rst [hunk 1]
```diff
@@ -190,9 +226,9 @@ permissions for newly created files and directories.
 
 The returned permissions are:
 
--  mode: Creation mode, like 0600.
+-  ``mode``: Creation mode, like 0600.
```
Comment (cmouse) [inline source/developer_manual/design/mailbox_list.rst:229]: definition list here too maybe?

File: source/developer_manual/design/mailbox_list.rst [hunk 2]
```diff
@@ -139,10 +173,10 @@ until it returns NULL. The returned mailbox_info struct contains:
       pattern.
 
       -  Example: "foo/bar" exists, but "foo" doesn't. "%", "foo" or
```
Comment (cmouse) [inline source/developer_manual/design/mailbox_list.rst:175]: It is better to use `` for escaping.

File: source/developer_manual/design/mailbox_transaction.rst [hunk 3]
```diff
@@ -33,6 +33,17 @@ flags:
    visible. You shouldn't usually need this, because usually you should
    have recently done a mailbox sync.
 
+-  ``MAILBOX_TRANSACTION_FLAG_NO_CACHE_DEC``: Don't update caching decisions
```
Comment (cmouse) [inline source/developer_manual/design/mailbox_transaction.rst:36]: definition list

**Review by cmouse — CHANGES_REQUESTED** [review]

**Comment by sirainen** [comment]
Added definition lists.

**Review by cmouse — APPROVED** [review]

**Pull request merged**